framework/src/modules/dpos_v2/commands/pom.ts

```diff
@@ -92,32 +99,41 @@ export class ReportDelegateMisbehaviorCommand extends BaseCommand {
 	// eslint-disable-next-line @typescript-eslint/no-empty-function
 	public async execute(context: CommandExecuteContext<PomTransactionParams>): Promise<void> {
 		const { chainID, getMethodContext, params, transaction, header } = context;
+
 		const currentHeight = header.height;
 		const header1 = BlockHeader.fromBytes(params.header1);
 		const header2 = BlockHeader.fromBytes(params.header2);
+
+		header1.validate();
+		header2.validate();
 		/*
 			|header1.height - h| >= 260,000.
 			|header2.height - h| >= 260,000.
 		*/
 
-		if (Math.abs(header1.height - currentHeight) >= MAX_PUNISHABLE_BLOCK_HEIGHT_DIFFERENCE) {
-			throw new Error(
-				`Difference between header1.height and current height must be less than ${MAX_PUNISHABLE_BLOCK_HEIGHT_DIFFERENCE.toString()}.`,
-			);
-		}
+		// if (Math.abs(header1.height - currentHeight) >= MAX_PUNISHABLE_BLOCK_HEIGHT_DIFFERENCE) {
+		// 	throw new Error(
+		// 		`Difference between header1.height and current height must be less than ${MAX_PUNISHABLE_BLOCK_HEIGHT_DIFFERENCE.toString()}.`,
+		// 	);
+		// }
+
+		// if (Math.abs(header2.height - currentHeight) >= MAX_PUNISHABLE_BLOCK_HEIGHT_DIFFERENCE) {
+		// 	throw new Error(
+		// 		`Difference between header2.height and current height must be less than ${MAX_PUNISHABLE_BLOCK_HEIGHT_DIFFERENCE.toString()}.`,
+		// 	);
+		// }
```

Comment:
If the old code is obsolete, it is safe to remove it, instead of keeping it commented out.

And if we need it in the future, we'll get it from Git 🙂

Reply:
As this is a draft PR, comments are ok and will be cleaned up in the final draft.

Reply:
@bobanm , cleaned up the comments.